src/app/data_harvester/memory/general/sysinfo.rs

```diff
@@ -8,13 +8,17 @@ pub async fn get_mem_data(
 ) -> (
     crate::utils::error::Result<Option<MemHarvest>>,
     crate::utils::error::Result<Option<MemHarvest>>,
+    crate::utils::error::Result<Option<MemHarvest>>,
 ) {
     use futures::join;
 
     if !actually_get {
-        (Ok(None), Ok(None))
+        (Ok(None), Ok(None), Ok(None))
+    } else if cfg!(feature = "zfs") && cfg!(target_os = "freebsd") {
+        join!(get_ram_data(sys), get_swap_data(sys), get_arc_data())
```

Comment:
Same nit as in `heim.rs`, personally prefer explicitly figuring this out on compilation. It might get removed anyways during compilation but it's just a preference to make it explicitly so.

Reply:
Removed this runtime check